docs/external_libraries.rst

```diff
@@ -0,0 +1,43 @@
+External libraries
+==================
+
+The Taichi Programming Language provides a set of built-in standard library.
+The Taichi built-in library is aimed to be simple for maintainability, and
+therefore generic and less domain specific.
+
+However users may find that the built-in library along may not enough,
+so here we provide some external libraries that hopefully could extend your
+productivity in Taichi.
+
+
+Taichi GLSL
+-----------
+
+`Taichi GLSL <https://github.com/taichi-dev/taichi_glsl>`_ is an external
+library of Taichi, aimed to provide a set of useful helper functions including
+but not limited to:
+
+1. Handy scalar functions like ``clamp``, ``smoothstep``, ``mix``, ``round``.
+2. GLSL-alike vector functions like ``normalize``, ``distance``, ``reflect``.
+3. Well-behaved random generators including ``randUnit2D``, ``randNDRange``.
+4. Possible Taichi BUG hotfixes that are not yet released in it's cycle.
```

Comment:
Maybe this is not really the job that an extension library should do?

Reply:
But that's what Taichi GLSL **have to do** before v0.6.11. Sometimes Taichi BUGs can limit Taichi GLSL's ability, e.g. docstring this time.